### ISSUE TITLE
Add server in whitelist mode

### ISSUE THREAD

**Issue by gigantuar** [DagAgren/toot#402]
I just purchased the app and am trying to add my server configured in whitelist mode. I am unable to do so through the add server page. Is there a special syntax or menu to add a server by address manually and not search the list of publicly fetched servers?

**Comment by DagAgren** [comment]
I think whitelist mode kind of breaks everything, if I recall correctly from some earlier discussion? I would not recommend it.

**Comment by rsmk2** [comment]
Just wanted to say that I have exactly the same problem. I run a server in whitelist mode for my family. Amaroq and Mast finds it without problem. Why doesn't Toot find it? (and my apologies if this has been asked before).

Anyway, if you find some way to fix this it would be great!

**Comment by gigantuar** [comment]
I ended up moving my private family instance off Mastodon and onto Pleroma with federation disabled. Toot is able to find my private Pleroma instance fine. I'm guessing Toot uses an API which is disabled in whitelist mode? Every other Mastodon/Pleroma app I tried worked perfectly fine in whitelist mode however.

**Comment by DagAgren** [comment]
Was it api/v1/instance, I think? That is needed for a lot of internal logic.

**Comment by noiob** [comment]
yes, that api endpoint gets turned off in limited federation mode
Here's what I patched to get an instance in that mode to look like a normal instance from the outside: https://github.com/noiob/awoospace/commit/17e40b4c9029806deec6d977017a75a3f8eea39e
